expertise/service/routes.py

```diff
@@ -0,0 +1,458 @@
+        flask.current_app.logger.info(f'Looking at {profile_dir}')
+
+        # Check for profile directory
+        if not os.path.isdir(profile_dir):
+            raise OpenReviewException('No jobs submitted since last server reboot')
+
+        # Check for error log
+        flask.current_app.logger.info(f'Checking error log')
+        err_dir = os.path.join(profile_dir, 'err.log')
+        ## Build list of jobs
+        if os.path.isfile(err_dir):
+            with open(err_dir, 'r') as f:
+                err_jobs = f.readlines()
+            err_jobs = [list(item.strip().split(',')) for item in err_jobs]
+            for id, name, err in err_jobs:
+                result['results'].append(
+                    {
+                        'job_id': id,
+                        'name': name,
+                        'status': f'Error: {err}'
+                    }
+                )
+
+        # Perform a walk of all job sub-directories for score files
+        job_subdirs = get_subdirs(profile_dir)
+        # If given an ID, only get the status of the single job
+        if job_id is not None:
+            job_subdirs = [name for name in job_subdirs if name == job_id]
+        flask.current_app.logger.info(f'Subdirs: {job_subdirs}')
+
+        for job_dir in job_subdirs:
+            search_dir = os.path.join(profile_dir, job_dir)
+            flask.current_app.logger.info(f'Looking at {search_dir}')
+            file_dir, _ = get_score_and_metadata_dir(search_dir)
+            # Load the config file to fetch the job name
+            with open(os.path.join(search_dir, 'config.cfg'), 'r') as f:
+                config = json.load(f)
+            flask.current_app.logger.info(f'Current score status {file_dir}')
+            # If found a non-sparse, non-data file CSV, job has completed
+            if file_dir is None:
+                status = 'Processing'
+            else:
+                status = 'Completed'
+            result['results'].append(
+                {
+                    'job_id': job_dir,
+                    'name': config['name'],
+                    'status': status
+                }
+            )
+                        
+    except openreview.OpenReviewException as error_handle:
+        flask.current_app.logger.error(str(error_handle))
+
+        error_type = str(error_handle)
+        status = 500
+
+        if 'not found' in error_type.lower():
+            status = 404
+        elif 'forbidden' in error_type.lower():
+            status = 403
+
+        result['error'] = error_type
+        return flask.jsonify(result), status
+
+    # pylint:disable=broad-except
+    except Exception as error_handle:
+        result['error'] = 'Internal server error: {}'.format(error_handle)
+        return flask.jsonify(result), 500
+        
+    else:
+        flask.current_app.logger.debug('POST returns ' + str(result))
+        return flask.jsonify(result), 200
```

Comment:
This is a minor thing but why is this outside the `try:` block?

Reply:
This gets executed only if the try block passes with no errors - this was just the way it was implemented in the original matcher code I pulled from.

Looking at it again, I could just put this code in the try block